ls1intum/Artemis
# [Exam] Management: Follow-up fixes

### Checklist

### Motivation and Context
This weeks new features for the exam management introduced a few smaller issues.

### Description

- [x] Ensured that exercise groups are preserved on exam update (changing exercise groups should be handled via the `ExerciseGroupResource`) (see: https://github.com/ls1intum/Artemis/pull/1664/commits/f0d9c3260c985cf4da3527d864781cb54836eb70)
- [x] Fixed missing exercise group order
- [x] Fixed `null index column for collection Exam.exerciseGroups` when getting the exam with eager exercise groups
- [x] Ensured that exercise groups cannot be manipulate via `@PostMapping("/courses/{courseId}/exams")`
- [x] @krusche @jpbernius: We've removed `TODO: check that the exam id in the body was NOT changed` as we have nothing to check against. Should we add `{examId}` to the route?
- [x] Exercise groups and associated exercises will also be deleted on deletion of exam
- [x] Exams will be deleted on deletion of course
- [x] Ensured that the not yet implemented *Assessment Dashboard* link in the exam overview does not throw an error
- [x] Used `ExamAccessService` checks in the following routes: `/courses/{courseId}/exams/{examId}/students` [POST], `/courses/{courseId}/exams/{examId}/students/{studentLogin}` [POST, DELETE]

### Steps for Testing
Mostly code review :-)

**Made sure that exercise groups are preserved on exam update**
Update an exam at check that it still works. We've covered the intended preserve with a test case.

**Fixed missing exercise group order:**
Test this locally:
1. Create a new exam
2. Add some exercise groups to the exam
3. Run this query on your local database: `SELECT * FROM exercise_group;`
4. Check that `exercise_group_order` is not null for the new exercise groups

**Exercise groups and associated exercises will also be deleted on deletion of exam**
1. Create a new Exam 
2. Add an exercise group
3. Add an exercise
4. delete the Exam

**Exams will be deleted on deletion of course**
1. Create a new Course
2. Add a new Exam 
3. Add an exercise group
4. Add an exercise
5. Delete the Course

**Fixed `null index column for collection Exam.exerciseGroups` when getting the exam with eager exercise groups**
1. Create a new exam
2. Add a new exercise group to the exam
3. Change the title of the exam
4. Check that the update works without an error on console
